### PR TITLE
Tower of Hanoi in Python

### PR DESCRIPTION
**Pull request by sayantani11**
Fixes issue #828 
Review and suggest if any changes required.

### CODE REVIEW

**Comment by sayantani11** [comment]
@raksha009 Please review

**Comment by sayantani11** [comment]
@plazzy99 @raksha009 please review

File: Code/Python/Tower_of_Hanoi.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    ---|---        |           |                   |           |        ---|---    
+   =========   =========   =========           =========   =========   =========   
+      A		   B		C		   A	       B	   C
+
+ Program will print the source and destination to transfer each disks for all the steps in chronological order.'''
+
+
+def Tower_of_Hanoi(n,src,dest,help):
+
+   ''' This function takes the number of disks and the names of the rods.
```
Comment (raksha009) [inline Code/Python/Tower_of_Hanoi.py:21]: Add this comment in brief in form of a single line comment above the function.

File: Code/Python/Tower_of_Hanoi.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+if __name__=='__main__':
+
+    n=int(input("Enter number of rods: "))
+
+    Tower_of_Hanoi(n,'A','B','C')
+
+
+''' Sample Input = 3
```
Comment (raksha009) [inline Code/Python/Tower_of_Hanoi.py:43]: Plz go through guidelines and pinned issues.
Follow
Test Cases
Input : 
Output: 
(Atleast 2)

Time and space Complexities.

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by sayantani11** [comment]
@raksha009 please check

File: Code/Python/Tower_of_Hanoi.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ Program will print the source and destination to transfer each disks for all the steps in chronological order.'''
+
+# This function takes the number of disks and the names of the rods.
+#	The idea is to shift one disk from the source to the destination but
+#	if there is already mutiple disk on the source then call the same function to put the extra disk on the helper rod and
+#	if there is already mutiple disk on the destination then call the same function to put the extra disk on the helper rod. 
+ 
+def Tower_of_Hanoi(n,src,dest,help):
+
+    if(n==0): # defining the base case of the function
```
Comment (raksha009) [inline Code/Python/Tower_of_Hanoi.py:25]: Add comments above the line

File: Code/Python/Tower_of_Hanoi.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        return
+
+    Tower_of_Hanoi(n-1,src,help,dest) # recursively calling the function
```
Comment (raksha009) [inline Code/Python/Tower_of_Hanoi.py:28]: remove this comment

File: Code/Python/Tower_of_Hanoi.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    # moving disk from source to helper rod
+
+    print(f"Move from {src} to {dest}")
+
+    Tower_of_Hanoi(n-1,help,dest,src) #moving disk from helper to destination rod
```
Comment (raksha009) [inline Code/Python/Tower_of_Hanoi.py:33]: Add comments above the line

File: Code/Python/Tower_of_Hanoi.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+#Driver code of the function
+if __name__=='__main__':
+
+    n=int(input("Enter number of rods: "))  # taking the number of rods from the user
```
Comment (raksha009) [inline Code/Python/Tower_of_Hanoi.py:38]: remove this comment

File: Code/Python/Tower_of_Hanoi.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+    Tower_of_Hanoi(n,'A','B','C')
+    # A B C are the name of rods
```
Comment (raksha009) [inline Code/Python/Tower_of_Hanoi.py:41]: Add comments above the line 40

File: Code/Python/Tower_of_Hanoi.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+''' 
+TESTCASES:
+
+Sample Input :
```
Comment (raksha009) [inline Code/Python/Tower_of_Hanoi.py:46]:
```suggestion
Input :
```

File: Code/Python/Tower_of_Hanoi.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+   Enter number of rods: 3
+
+Sample Output:
```
Comment (raksha009) [inline Code/Python/Tower_of_Hanoi.py:49]:
```suggestion
Output:
```

And do this change in other i/p - o/p

File: Code/Python/Tower_of_Hanoi.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+Sample Input:
+   Enter number of rods: 2
+
+Sample Output:
+   Move from A to C
+   Move from A to B
+   Move from A to C
+
+   Time Complexity : O(2^n)
+   Space Complexity : O(n)
```
Comment (raksha009) [inline Code/Python/Tower_of_Hanoi.py:69]: specify n below line 69.

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by sayantani11** [comment]
@raksha009

**Review by raksha009 — APPROVED** [review]

**Pull request merged**